### ISSUE TITLE
Darček <GIFT> pre Heuréku

### ISSUE THREAD

**Issue by slavomlynar** [pavelevap/ceske-sluzby#180]
Zdravím,
potrebujem poradiť ako cez woocommerce + české služby,  môžem pridať darček k produktu + aby sa zobrazoval pri exporte xml na heuréke

**Comment by pavelevap** [comment]
@slavomlynar: Asi není problém podobné políčko přidat, ale spíše přemýšlím kam :-) Jak to budete chtít nastavovat? Ke každému produktu máte nějak uvedený/nastavený produkt zdarma? Nebo chcete mít dárek zdarma pro každý produkt a nastavit to hromadně pro celý eshop?

**Comment by slavomlynar** [comment]
Ďakujem za vašu reakciu.
Myslím že cez položku ( produkt/Vlasnosti/Uživatelom zadaná vlastnosť/ meno - hodnota a viditelné na stránke produktu) alebo iným spôsobom. Nechcem to nastaviť globálne. Chcem to dať iba k niektorým produktom.

**Comment by slavomlynar** [comment]
Skúšal som to cez plugin "Free Gift Options" ale tento plugin to pridá iba v košíku a neexportuje to do xml pre heureku. Myslel som že takto by to bolo jednoduché pre vkladanie a určovanie [http://test2.mlynar.org/wp.png](url)

**Comment by lukasprelovsky** [comment]
GIFT aktualne xml heureka CS sluzby nepodporuje a taky bych to ocenil 👍

**Comment by pavelevap** [comment]
A ten dárek zdarma bude tedy spíše vložený samostatný text (v podobě vlastnosti) nebo nějaký konkrétní produkt? Teoreticky by asi zatím bohatě stačilo, abychom podporovali zmiňovaný plugin a použili jeho nastavení (a posílali bychom to do XML feedu)?

**Comment by slavomlynar** [comment]
Nechám to na vás ako by to bolo vhodnejšie. Mne sa jedná v prvom rade o to aby som na heureke ponúkol niečo viac ako konkurencia pri rovnakom produkte, no zároveň nech je to uvedené a viditeľné aj pri produkte už priamo na eshope, aby to motivovalo kúpiť ten konkrétny produkt.

**Comment by pavelevap** [comment]
Na mě to určitě nenechávejte, používat to zatím nebudu :-) Možností je více, hlavní je asi mít dárek v podobě samostatného produktu a (přiřazovat ho k produktům) nebo tam mít nějaké políčko, kam se natvrdo dopíše, co je přesně dárek zdarma. Nevím, co by asi nejvíce vyhovovalo? @lukasprelovsky  

Ten plugin to umí zobrazit i přímo u produktu nebo ne? To je myslím podmínka Heureky, aby to tam bylo na stránce s produktem vidět.

**Comment by slavomlynar** [comment]
Mne by postačovalo keby to bolo cez vlastnosť, ktorá je viditeľná pri produkte a natvrdo to napíšem k produktu ku ktorému to chcem dať. (K príkladu uvádzam tlačiareň + darček https://multifunkcne-zariadenia.heureka.sk/epson-l655/#  a následne pri prekliku na eshop je darček uvedený len v nadpise produktu http://www.epsonshop.sk/epson-l655 ). Hlavne nech to vidí heureka a ja si ten darček rozpíšem pri produkte, alebo tú vlastnosť si dám zobraziť (je do zaškrtávacie políčko pri vlastnostiach (zobraziť na stránke produktu)

**Comment by slavomlynar** [comment]
Ten plugin ktorý som vyskúšal funguje správne, ale funguje iba až po vstupe do košíka a až vtedy dá vybrať si darček (mám to zatial nastavené na eshope https://shop.mckero.eu/produkt/supremio-komplet-set/ po pridaní tohto produktu a až po vstupe do košíka ponúkne darček)

**Comment by lukasprelovsky** [comment]
Kdyz koukam https://sluzby.heureka.sk/napoveda/xml-feed/#GIFT tak pro mne by bylo nejlepsi asi manualne vypsat do title + darek a u produktu v zalozce ceske sluzby bych si mohl definovat darek.

Nevim zda je vhodny instalovat jeste dalsi plugin.

ja bych to k vlastnostem asi nechtel (nevim jestli to je rozumny takhle), ale osobne si myslim ze by stacilo:

1.manualne napsat k popisu produktu : 
Nazov produktu + Darek 500gb hardisk
2. U editaci produktu v zalozce ceske sluzby by bylo pole Darek k produktu, tam by byla moznost pridani vice darku. U kazdeho cisla darku by bylo mozne definovat nazev. A tohle by se uverejnovalo v heureka XML.

Nebo si to jednoduse jenom predstavuji jak to heureka asi chce? :)

**Comment by slavomlynar** [comment]
Len neviem keby som dal do názvu produktu "+ darček" či by správne spárovalo produkt do heuréky. Podla mňa je najlepšie mať čistý a presný názov produktu. 
Definícia podľa heureky: Presný názov produktu. Nesmie obsahovať žiadne iné informácie, ako napr. + darček, alebo doprava zdarma. Hodnota musí byť obsiahnutá tiež v tagu PRODUCT, inak nebude braný v úvahu. Pokiaľ názvy produktov nebudú obsahovať produktové číslo, hrozí, že nedokážeme produkty správne identifikovať a tiež zaradiť. Maximálna dĺžka názvu produktu je 255 znakov. 
https://sluzby.heureka.sk/napoveda/xml-feed/#PRODUCTNAME

**Comment by pavelevap** [comment]
Potom je ten dárek v názvu spíše element `PRODUCT`? Ten už umíme, takže je možné ho doplnit: #130

**Comment by slavomlynar** [comment]
Momentálne plugin česke služby exportuje názov produktu do PRODUCTNAME, neviem ako to majú ostatné eshopy, ale heureka požaduje parameter GIFT

**Comment by slavomlynar** [comment]
Možno stačí dodať len jedno textové pole do pluginu kde sa zadávajú parametre pre heureku

**Comment by pavelevap** [comment]
Ano, ale pokud tomu dobře rozumím, tak do `GIFT` by se dal nějaký navolený produkt nebo speciální popisek (ten prostě musíme posílat) a ještě by se mohl pomocí `PRODUCT` upravit celý název produktu (jenom obecně "+ dárek" nebo něco podobného).

**Comment by slavomlynar** [comment]
Áno presne tak

**Comment by lukasprelovsky** [comment]
exemple : https://www.inhodinky.sk/festina-damske-hodinky/festina-164921-darcek/

ten vyber darku se mi libi, ale neznam plugin na to :)

**Comment by pavelevap** [comment]
Začíná to dávat smysl, zkusím něco vymyslet. Název produktu je tedy asi jasný, podporovat můžeme definovanou vlastnost nebo i výběr produktů, otázkou je, zda v tomto případě pouze podpořit nějaký existující plugin nebo přidat vlastní políčko na výběr produktů :-) V případě, že bude `GIFT` jen text (či vlastnost), tak to není problém, ale pokud to budou produkty, tak budeme muset řešit i jejich přidání do košíku a do objednávky? Nebo to tam být nemusí? Bude ale třeba je pak zobrazovat na stránce s produktem. Ještě se podívám, jaké na to existují pluginy...

**Comment by slavomlynar** [comment]
V prípade že darček bude niektorý produkt tak to bude musieť ísť aj do košíka z dôvodu množstva na sklade , aby to dopočítalo zo skladu

**Comment by Suffokate** [comment]
zaplatim za dokodenie inputu pre parameter gift nepotrebujeme aby sa to nejak parovalo s produktami ide nam cisto len to textovy input. Ak by si na to niekto trufal, moj mail je marek(at)smartzero(dot)sk. Ďakujem
